WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift

```diff
@@ -51,6 +52,19 @@ struct GutenbergNetworkRequest {
                     case .success(let response):
                         completion(.success(response))
                     case .failure(let error):
+                        if path.starts(with: "/oembed/1.0/") {
+                            if let error = error as? AFError {
+                                if case .responseValidationFailed(let reason) = error {
+                                    if case .unacceptableStatusCode(let code) = reason {
+                                        if code == 404 {
+                                            let errorCode = URLError.Code(rawValue: code)
+                                            completion(.failure(URLError(errorCode) as NSError))
+                                            return
+                                        }
+                                    }
+                                }
+                            }
+                        }
```

Comment:
Does this mean that the HTTP status code is not included in the error object? If so, probably we should revisit this part in the future because having error codes on the RN side could be very helpful.

On the other hand, I'm wondering if we could have this logic in another function as it looks very specific to the embed case, wdyt?

Reply:
> Does this mean that the HTTP status code is not included in the error object? If so, probably we should revisit this part in the future because having error codes on the RN side could be very helpful.

I think the whole stack for error was included in the object returned to JS, but that doesn't have a `code` field that includes the response code for the case of self hosted site using rest api. I wasn't sure if returning the code could cause any unwanted side effects for other calls, so I just kept it specific to oembed path for now.

> On the other hand, I'm wondering if we could have this logic in another function as it looks very specific to the embed case, wdyt?

Yeah, we can do that 👍

Reply:
> I think the whole stack for error was included in the object returned to JS, but that doesn't have a code field that includes the response code for the case of self hosted site using rest api. I wasn't sure if returning the code could cause any unwanted side effects for other calls, so I just kept it specific to oembed path for now.

Ok, makes sense 👍 , I'm wondering if we should open an issue as a follow-up so we could revisit it in the future.